x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/common/ProgressListenableActionFuture.java

```diff
@@ -189,6 +200,16 @@ private static void executeListener(final ActionListener<Long> listener, final S
         }
     }
 
+    private static void safeAcceptProgress(LongConsumer consumer, long progress) {
+        assert consumer != null;
+        try {
+            consumer.accept(progress);
+        } catch (Exception e) {
+            assert false : e;
+            throw e;
```

Comment:
I'd probably prefer to log a warning instead rather than rethrow. There seems to be no reason to throw out the exception if the consumer were to fail (I am sure it will not for now).

Reply:
Sure, I pushed b2249a64940c5e19519002d0349f0a6b6da95513. I don't expect this to throw though.